src/main/java/build/buildfarm/instance/memory/DelegateCASMap.java

```diff
@@ -60,16 +74,24 @@ public V get(Object key) {
   }
 
   public boolean containsKey(Object key) {
-    return digestMap.get(key) != null;
+    return digestMap.containsKey(key) || emptyCache.getIfPresent(key) != null;
   }
 
   public V remove(Object key) {
     Digest valueDigest = digestMap.remove(key);
+    if (valueDigest == null) {
+      emptyCache.invalidate(key);
+      valueDigest = digestUtil.empty();
```

Comment:
But if it's something that wasn't in the map, aren't you supposed to return null?

Reply:
Too right. updated.